Integrate Flow for type annotations

mikkom opened the pull request:
This change integrates Flow for static type checking. It adds flow type definitions via flow-typed, required type annotations, and type definitions for container component props. The flow-react-proptypes Babel plugin is used to generate proptypes for runtime checks. Includes few other small fixes, along with two simple unit tests regarding proptype generation.

Resolves #22 and also #44

garetmckinley commented:
Hello @mikkom, thanks a lot for the contribution, I really appreciate it 😸 

Going to review it now 👍  I'm curious what exactly fixed #44?

garetmckinley commented on yarn.lock:
```diff
@@ -0,0 +1,3846 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
I'm a bit weary to re-introduce this, as we originally removed the lockfile in #9. But now that I read this article, published after that PR (https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all) I feel a little better about the possibility of including it. 

I wonder if it would be worthwhile to possibly add this to the `.npmignore` to remove the bulk of almost 3k lines to the package size. I'll do some research on it.

mikkom commented:
Hi, thanks for all comments! :)  I think it may have been just the version update of babel-presets and/or eslint-plugin-import that fixed #44. I can check once I'm back at my computer. I'll make the requested changes then, as well. Cheers!

mikkom commented:
Ok, I now added yarn.lock to .npmignore and updated .travis.yml to run the test script (ESLint, Flow & unit tests).

And it seems that it was just the updating of babel-preset-es2015-rollup to latest version that fixed #44.

garetmckinley commented:
Thanks a lot for the work you put in on this 👍
